Repository URL support. (#305)
MicrosoftDocs/vsonline: vs-online/reference/repository.md

---
author: edgonmsft
ms.author: edgonmsft
ms.service: visual-studio-online
title: Repository Options Visual Studio Online
ms.topic: overview
ms.date: 02/14/2020
---

# Repository support

> Visual Studio Online's [environments](../overview/what-is-vsonline.md#environments) can be initialized with a git repository. A [devcontainer.json](configuring.md) file contained inside the repository also provides additional configuration opportunities.

Reviewer: ```suggestion
Visual Studio Online's [environments](../overview/what-is-vsonline.md#environments) can be initialized with a Git repository. A [devcontainer.json](configuring.md) file contained inside the repository also provides additional configuration opportunities.
```

> Repository support is handled using the git clone url obtained from your source control provider.

Reviewer: ```suggestion
Repository support is handled using the Git clone url obtained from your source control provider.
```

Any URL that can be cloned directly locally on your machine should work in this way.

## Private Repositories

> ### VSCode

Reviewer: ```suggestion
### Visual Studio Code extension
```

> Our Visual Studio Code extension uses your local git installation to obtain credentials to perform clone operations. We do not save your credentials anywhere.

Reviewer: ```suggestion
Our Visual Studio Code extension uses your local Git installation to obtain credentials to perform clone operations. We do not save your credentials anywhere.
```

Support is straight forward, if you can perform operations locally they should just work.

### Browser

> Our browser support performs OAuth flows for the selected provider of the git url provided. We are continuously working to improve and extend our support for additional providers.

Reviewer: ```suggestion
Our browser support performs OAuth flows for the selected repository hosting provider of the Git url given. We are continuously working to improve and extend our support for additional hosting providers.
```

> For now we support the following:

Reviewer: ```suggestion
We currently support the following:
```
